picard/file.py

```diff
@@ -511,6 +511,7 @@ def _lookup_finished(self, lookuptype, document, http, error):
         if self.state == File.REMOVED:
             return
 
+        self.clear_pending()
```

Comment:
Why is this necessary? AFAICT all paths through this function call `clear_pending` already.